libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -164,10 +180,55 @@ async def acall_tool(state: AgentState, config: RunnableConfig):
     return workflow.compile()
 
 
+def get_model_runnable_from_state_modifier(
+    state_modifier: Optional[StateModifier], model: LanguageModelLike
+) -> Runnable:
+    if state_modifier is None:
+        model_runnable = (lambda state: state["messages"]) | model
+    elif isinstance(state_modifier, str):
+        _system_message: BaseMessage = SystemMessage(content=state_modifier)
+        model_runnable = (lambda state: [_system_message] + state["messages"]) | model
+    elif isinstance(state_modifier, SystemMessage):
+        model_runnable = (lambda state: [state_modifier] + state["messages"]) | model
+    elif isinstance(state_modifier, types.FunctionType):
+        model_runnable = (lambda state: state_modifier(state)) | model
+    elif isinstance(state_modifier, Runnable):
+        model_runnable = state_modifier | model
+    else:
+        raise ValueError(
+            f"Got unexpected type for `state_modifier`: {type(state_modifier)}"
+        )
+
+    return model_runnable
+
+
+def convert_messages_modifier_to_state_modifier(
```

Comment:
@hwchase17 added this helper, so should be easy to handle the messages_modifier-style functions from langchain